### ISSUE TITLE
Wrong behavior when ≥2 tables in blockquote (>)

### ISSUE THREAD

**Issue by playermiller109** [NicoNekoru/obsidan-advanced-table-xt#45]
Hi, when writing tables in blockquote, they will join and repetitive display.

Here is test image, left: reading mode; right: source mode.

<img width='400' src='https://github.com/NicoNekoru/obsidan-advanced-table-xt/assets/145541890/bab76bcc-f4eb-42de-bcec-3f5d7c26ff19)'>

The test text:

```
>| | |
>| ---- | ---- |
>| TEST | 1 |
>
>| | |
>| ---- | ---- |
>| TEST | 2 |
```

I have tested in sandbox vault. My Sheets Extended Plugin version is 1.2.5，Obsidian Current and Installer version are v1.5.3，Default Theme.

Thanks.

**Comment by NicoNekoru** [comment]
Should be fixed by deafc15

**Comment by playermiller109** [comment]
@NicoNekoru

The reason for this issue is that when the table is in blockquote `>`, the `lineStart` and `lineEnd` are not those of the table, but of the entire block.

Therefore, if `.findIndex()` is used to search, and if `>` is a callout, it will find the head of the callout, which is the line `>[!note]` for example.

As a result, `textContent.slice(0, endIndex + 1)` will cut off all tables, causing them not to display.

Additionally, if there are two or more tables in `>`, `.findIndex()` will always find the position of the first table, causing all tables to render as the first one.

<image width="500" src="https://github.com/NicoNekoru/obsidan-advanced-table-xt/assets/145541890/6210abcb-2b92-4245-a488-61b55310134f">

(Sheets Extended Plugin version is 1.2.10, Obsidian Current and Installer versions are v1.5.12)
